numpy/fft/_pocketfft.py

```diff
@@ -1350,8 +1350,30 @@ def rfft2(a, s=None, axes=(-2, -1), norm=None):
         Input array, taken to be real.
     s : sequence of ints, optional
         Shape of the FFT.
+
+        .. versionchanged:: 2.0
+
+            If it is ``-1``, the whole input is used (no padding/trimming).
+
+        .. deprecated:: 2.0
+
+            If `s` is not ``None``, `axes` must not be ``None`` either.
+
+        .. deprecated:: 2.0
+
+            `s` must contain only ``int`` s, not ``None`` values. ``None``
+            values currently mean that the default value for ``n`` is used
+            in the corresponding 1-D transform, but this behaviour is
+            deprecated.
```

Comment:
The last deprecation directive has a lot of detail - would it be possible to condense them into one and shorten the second description? Something like (assuming I've understood correctly):

```suggestion
            If `s` is not ``None``, `axes` must not be ``None`` either.

            If `s` is a sequence, it must not contain `None`.
```

Reply:
I deliberately included the detail in the message since the behaviour with `None` values is undocumented, see gh-14179.

But I'm happy to change to your suggestion if that detail is unnecessary!

Reply:
Sounds reasonable to me!